pkg/service/backup/worker_deduplicate.go

```diff
@@ -134,48 +135,55 @@ func (w *worker) deduplicateUUIDSStables(remoteSSTables, localSSTables *sstableB
 	return deduplicated
 }
 
+// willCreateVersioned corresponds to snapshotDir.willCreateVersioned.
 func (w *worker) deduplicateIntSSTables(ctx context.Context, host string, remoteDir, localDir string,
 	remoteSSTables, localSSTables *sstableBundlesByID,
-) (deduplicated []fileInfo, err error) {
+) (deduplicated []fileInfo, willCreateVersioned bool, err error) {
 	// Reference to SSTables 3.0 Data File Format
 	// https://opensource.docs.scylladb.com/stable/architecture/sstable/sstable3/sstables-3-data-file-format.html
 
 	// Per every SSTable files group, compare local <ID>-Digest.crc32 content
 	// to the remote <ID>-Digest.crc32 content.
 	// The same content implies that SSTable can be deduplicated and removed from local directory.
 	for id, localBundle := range localSSTables.intID {
+		remoteBundle, ok := remoteSSTables.intID[id]
+		if !ok {
+			continue
+		}
+		// At this point analyzed SSTable ID is present in both local and remote dirs.
+		// Not being able to deduplicate it results in setting 'willCreateVersioned' to true.
 		crc32Idx := slices.IndexFunc(localBundle, func(fi fileInfo) bool {
 			return strings.HasSuffix(fi.Name, "Digest.crc32")
 		})
 		if crc32Idx == -1 {
+			willCreateVersioned = true
```

Comment:
Why it will create versioned if there is no crc32 file ? We don't know it. It will create versioned only if the checksum generated by Rclone are different.
But there is no option to validate it at this stage, so yeah it's possible better to assume it will.

Reply:
That's exactly my though process:)